diff --git a/kernel/Kconfig.hz b/kernel/Kconfig.hz
index 2a202a84..539d3e4e 100644
--- a/kernel/Kconfig.hz
+++ b/kernel/Kconfig.hz
@@ -3,9 +3,9 @@
 #
 
 choice
 	prompt "Timer frequency"
-	default HZ_250
+	default HZ_600
 	help
 	 Allows the configuration of the timer frequency. It is customary
 	 to have the timer interrupt run at 1000 Hz but 100 Hz may be more
 	 beneficial for servers and NUMA systems that do not need to have
@@ -38,8 +38,16 @@ choice
 	 while also showing good interactive responsiveness even
 	 on SMP and NUMA systems and exactly dividing by both PAL and
 	 NTSC frame rates for video and multimedia work.
 
+	config HZ_600
+		bool "600 HZ"
+	help
+	 600 Hz is a balanced timer frequency. Provides fast interactivity
+	 on desktops with great smoothness without increasing CPU power
+	 consumption and sacrificing the battery life on laptops.
+	 N.B.: Originally it was 500 Hz, but this has been modified.
+
 	config HZ_1000
 		bool "1000 HZ"
 	help
 	 1000 Hz is the preferred choice for desktop systems and other
@@ -51,8 +59,9 @@ config HZ
 	int
 	default 100 if HZ_100
 	default 250 if HZ_250
 	default 300 if HZ_300
+	default 600 if HZ_600
 	default 1000 if HZ_1000
 
 config SCHED_HRTICK
 	def_bool HIGH_RES_TIMERS
